amundsen_application/static/js/components/common/AvatarLabel/index.tsx

```diff
@@ -1,19 +1,32 @@
 import * as React from 'react';
 import * as Avatar from 'react-avatar';
 
+import { Link } from 'react-router-dom';
+import { logClick } from 'ducks/utilMethods';
+
 // TODO: Use css-modules instead of 'import'
 import './styles.scss';
 
 export interface AvatarLabelProps {
   label?: string;
   src?: string;
+  link?: string;
+  target?: string;
 }
 
-const AvatarLabel: React.SFC<AvatarLabelProps> = ({ label, src }) => {
+const AvatarLabel: React.SFC<AvatarLabelProps> = ({ label, src, link, target }) => {
   return (
     <div className='avatar-label-component'>
       <div id='component-avatar' className='component-avatar'>
-        <Avatar name={label} src={src} size={24} round={true} />
+        <Link
+          to={ link }
+          target={ target }
+          className="avatar-overlap"
+          id="frequent-users"
+          onClick={logClick}
+        >
+          <Avatar name={label} src={src} size={24} round={true} />
+        </Link>
       </div>
       <label id='component-label' className='component-label'>{label}</label>
     </div>
```

Comment:
- For other usages of the `<AvatarLabel>`, we have been wrapping it in a `<Link>` like so (see TableDetails/index.tsx)
```
<a href={appUrl} target='_blank' id="explore-writer" onClick={logClick}>
  <AvatarLabel label={appName} src={sourceImage}/>
</a>
```
If we're going to build links into the `<AvatarLabel>` component we might as well update the other usages as well. That being said, we've done quite a bit of refactoring in [table_V2 feature branch](https://github.com/lyft/amundsenfrontendlibrary/tree/feature/table_v2) so it might not make sense to update that now.
- I don't think it's safe to use `id="frequent-users"` since this is a generic component and not made specifically for the `frequent-users`. As a side note, the `logClick` method will pass the target element's ID as a logging parameter.
- Why not make the label on line 31 clickable as well? You'd just need to move it inside the `<Link>` component.

Reply:
@danwom thank you for these findings!

Note that I've still left additional fields at AvatarLabelProps. If you feel that it may be conceptually incorrect - please let me know